starter-code/index.js

```diff
@@ -1,17 +1,56 @@
-class SortedList {
-  constructor() {}
-
-  add(item) {}
-
-  get(pos) {}
-
-  max() {}
-
-  min() {}
-
-  sum() {}
-
-  avg() {}
-}
-
-module.exports = SortedList;
+class SortedList {
+  constructor(items, length) {
+    this.items = [];
+    this.length = this.items.length;
+  }
+
+  add(item) {
+    this.items.push(item);
+    this.length = this.items.length;
+    this.items.sort((a, b) => a - b);
+  }
+
+  get(pos) {
+    if (pos > this.length) {
+      throw new Error("OutOfBounds");
+    } else {
+      return this.items[pos];
+    }
+  }
+
+  max() {
+    if (this.length === 0) {
+      throw new Error("EmptySortedList");
+    } else {
+      return Math.max(...this.items);
+    }
+  }
+
+  min() {
+    if (this.length === 0) {
+      throw new Error("EmptySortedList");
+    } else {
+      return Math.min(...this.items);
+    }
+  }
+
+  sum() {
+    if (this.length === 0) {
+      return 0;
+    } else {
+      return this.items.reduce((accu, value) => accu + value);
+    }
+  }
+
+  avg() {
+    if (this.length === 0) {
+      throw new Error("EmptySortedList");
+    } else {
+      return (
+        this.items.reduce((accu, value) => accu + value) / this.items.length

```

Comment:
Keep in mind that you can use the function: this.sum() so you don't have to reduce twice